Assets/Scripts/Selector.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Selector : MonoBehaviour
+{
+    private Queue<Tile> selectedElements;
+    [SerializeField] private List<Tile> selectedElementsList;
+
+    [SerializeField] private Tile previousTile;
+
+    public static event Action<Queue<Tile>> moveTiles;
+
+    void Start()
+    {
+        selectedElements = new Queue<Tile>();
+        selectedElementsList = new List<Tile>();
+    }
+
+    void Update()
+    {
+        selectedElementsList = new List<Tile>(selectedElements); // this is just a hack. Will probably delete later
+    }
+
+    private void Select(Tile tile)
+    {
+        if ((selectedElements.Count > 0) && !CanBeAddedToCurrentChain(tile))
+        {
+            if (IsMatch(tile))
+            {
+                Debug.Log("Bam");
+                foreach (Tile t in selectedElements)
+                {
+                    Debug.Log($"Destroyed tile {t.name}");
+                    //t.transform.position = new Vector2(0, 0);
+                }
+
+                moveTiles?.Invoke(selectedElements);
+            }
+
+            foreach (Tile t in selectedElements)
+            {
+                t.IsSelected = false;
+            }
+
+            selectedElements = new Queue<Tile>();
+        }
+
+        selectedElements.Enqueue(tile);
+        tile.IsSelected = true;
+        previousTile = tile;
+    }
+
+    private bool IsMatch(Tile tile)
+    {
+        return tile == previousTile && selectedElements.Count >= 3;
+    }
+
+    private bool IsAdjecent(Tile first, Tile second)
+    {
+        return ((second.Position.x == first.Position.x + 1 && second.Position.y == first.Position.y) ||
```

Comment:
for better readability - is it ok if we have variables named on top - with names like secondX,secondY and so on?